junit/core/src/test/java/org/jboss/arquillian/junit/MethodRuleChain.java

```diff
@@ -0,0 +1,45 @@
+package org.jboss.arquillian.junit;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+public class MethodRuleChain implements MethodRule {
```

Comment:
Can't we simply use `RuleChain` here? I'm not sure why we need this class. Especially that the way how we use it will result in calling our rule twice as it's part of the test and only then added to this rule chain (and they evaluated again).

Reply:
Cuurently `ArquillianTest` implements `MethodRule`. While `RuleChain` needs parameter of type `TestRule`. So we can't simply use `RuleChain` here. I will update the tests how should we use it as outer rule

Reply:
Let's expose it as a regular class so other users can leverage it too and add helpful JavaDoc to it.